Sandbox/BubbleSortLayer.cpp

```diff
@@ -0,0 +1,78 @@
+#include "BubbleSortLayer.h"
+
+#include <chrono>
+#include <thread>
+
+BubbleSortLayer::BubbleSortLayer()
+        : Aquarius::Layer("Bubble Sort") {}
+
+void BubbleSortLayer::onCreation()
+{
+    Aquarius::Application* app = Aquarius::Application::get();
+    Aquarius::Window* window = app->getWindow();
+
+    int height = window->getHeight();
+    int width = window->getWidth();
+
+    m_Camera = std::make_shared<Aquarius::OrthographicCamera>(1, 0.01, height, width);
+
+    Aquarius::Renderer::Init();
+
+    // Populate the bar heights with random values
+    for (int i = 0; i < 100; i++)
+    {
+        barHeights[i] = rand() % 750 + 30;
+    }
+}
+
+void BubbleSortLayer::onUpdate(Aquarius::timeDelta_t ts)
+{
+    m_Camera->onUpdate(ts);
+
+    Aquarius::Renderer::BeginScene(m_Camera.get());
+    Aquarius::Renderer::ClearColor({ 0.2, 0.3, 0.7 });
+    Aquarius::Renderer::Clear();
+
+
+    renderBars(100);
+
+    swapBars(i, j);
+    i = i+1;
+    j = j+1;
+
+    if(i == 99 and j ==100)
```

Comment:
Could use += here

Reply:
Also, at least on MSVC this is giving me a compiler error, and I had to change to `&&`.